hosts/ev-wistia.php

```diff
@@ -0,0 +1,240 @@
+        ),
+        array(
+          'id' => 'developer_key',
+          'label' => 'API Token',
+          'required' => true,
+          'explanation' => 'This needs to be generated in your Wistia account'
+        )
+      ),
+      'introduction' => "Wistia's API requires you to generate an API token from your account, in order to access your videos from another site (like this one).",
+      'api_url' => 'https://wistia.com',
+      'api_link_title' => 'Wistia'
+    );
+
+    update_option( 'sp_external_videos_options', $options );
+
+  }
+
+  public static function echo(){
+    $class = get_class();
+    echo 'Class is ' .  $class . '<br />';
+    $hostname = preg_split( "/SP_EV_/", $class, 2, PREG_SPLIT_NO_EMPTY );
+    print_r($hostname);
+    return;
+  }
+
+  /*
+  *  remote_author_exists
+  *
+  *  Requires appropriate method for THIS HOST
+  *  Used by SP_External_Videos::remote_author_exists()
+  *  Checks if remote author exists on this host
+  *
+  *  @type  function
+  *  @date  31/10/16
+  *  @since  1.0
+  *
+  *  @param   $host_id, $author_id, $developer_key
+  *  @return  boolean
+  */
+
+  public static function remote_author_exists( $host_id, $author_id, $developer_key ){
+
+    $url = "https://api.wistia.com/v1/account.json";
+    $headers = array( 'Authorization' => 'Basic ' . base64_encode( "api:$developer_key" ) );
+    $args['headers'] = $headers;
+
+    $response = wp_remote_request( $url, $args );
+    $code = wp_remote_retrieve_response_code( $response );
+    $body = json_decode( wp_remote_retrieve_body( $response ), true );
+
+    // return false on error
+    if( !$response || is_wp_error( $response ) || preg_match('/^[45]/', $code ) ) {
+      return false;
+    }
+
+    // for wistia: also check that this api key belongs to this user account
+    $userUrl = $body['url'];
+    $expectUrl = "http://$author_id.wistia.com";
+
+    if ( $userUrl != $expectUrl ) {
+      return false;
+    }
+
+    return true;
+
+  }
+
+  /*
+  *  embed_code
+  *
+  *  Used by save_video()
+  *  Embed code is stored as postmeta in external-video posts.
+  *  Code is specific to each host site's embed API.
+  *
+  *  @type  function
+  *  @date  31/10/16
+  *  @since  0.23
+  *
+  *  @param   $video_id
+  *  @return  <iframe>
+  */
+
+  public static function embed_code( $video_id ) {
+
+    return "//fast.wistia.net/embed/iframe/$video_id";
+
+  }
+
+  /*
+  *  fetch
+  *
+  *  WISTIA DATA API v1
+  *  check https://wistia.com/doc/data-api#making_requests
+  *
+  *  @type  function
+  *  @date  31/10/16
+  *  @since  1.0
+  *
+  *  @param   $author
+  *  @return  $new_videos
+  */
+
+  static function fetch( $author ) {
+
+    $author_id = $author['author_id'];
+    $developer_key = $author['developer_key'];
+
+    // set other options
+    $page = 0;
+    $per_page = 1;  // One by one because we get no next-page info from Wistia
+    $thumb_w = get_option( "large_size_w" ); // Size could be set in options at some point
```

Comment:
large_size_w and large_size_h are not currently defined - these should be hard coded to 180x135 (also note that the "X" below is capitalised, so creates an error with wistia.